lib/FilesHooks.php

```diff
@@ -758,6 +772,42 @@ protected function shareByLink($fileSource, $itemType, $linkOwner) {
 		);
 	}
 
+	/**
+	 * Sharing a file or folder with a team
+	 *
+	 * @param string $shareWith
+	 * @param int $fileSource File ID that is being shared
+	 * @param string $itemType File type that is being shared (file or folder)
+	 * @param string $fileTarget File path
+	 * @param int $shareId The Share ID of this share
+	 */
+	protected function shareWithTeam(string $shareWith, int $fileSource, string $itemType, string $fileTarget, int $shareId, string $sharer): void {
+		if ($this->teamManager === null) {
+			return;
+		}
+
+		try {
+			$this->teamManager->startSuperSession();
```

Comment:
What does it do? Does it needs to be stoped?

Reply:
Start the session. It allows to query also users that could not be detected by current user due to access restrictions (we want to inform all members so we need a session without a current user set).

https://github.com/nextcloud/circles/blob/08a08b12e2ca8207a2d8fa09a410363ace9b4719/lib/CirclesManager.php#L168

Reply:
(I never saw that the session is stopped in collectives or other apps that use teams)